File: 00_Alternate_Languages/76_Russian_Roulette/javascript/russianroulette.js
// RUSSIAN ROULETTE
//
// Converted from BASIC to Javascript by Oscar Toledo G. (nanochess)
//

function print(str)
{
    document.getElementById("output").appendChild(document.createTextNode(str));
}

function input()
{
    var input_element;
    var input_str;

    return new Promise(function (resolve) {
                       input_element = document.createElement("INPUT");

                       print("? ");
                       input_element.setAttribute("type", "text");
                       input_element.setAttribute("length", "50");
                       document.getElementById("output").appendChild(input_element);
                       input_element.focus();
                       input_str = undefined;
                       input_element.addEventListener("keydown", function (event) {
                                                      if (event.keyCode == 13) {
                                                      input_str = input_element.value;
                                                      document.getElementById("output").removeChild(input_element);
                                                      print(input_str);
                                                      print("\n");
                                                      resolve(input_str);
                                                      }
                                                      });
                       });
}

function tab(space)
{
    var str = "";
    while (space-- > 0)
        str += " ";
    return str;
}

// Main program
async function main()
{
    print(tab(28) + "RUSSIAN ROULETTE\n");
    print(tab(15) + "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n");
    print("\n");
    print("\n");
    print("\n");
    print("THIS IS A GAME OF >>>>>>>>>>RUSSIAN ROULETTE.\n");
    restart = true;
    while (1) {
        if (restart) {
            restart = false;
            print("\n");
            print("HERE IS A REVOLVER.\n");
        }
        print("TYPE '1' TO SPIN CHAMBER AND PULL TRIGGER.\n");
        print("TYPE '2' TO GIVE UP.\n");
        print("GO");
        n = 0;
        while (1) {
            i = parseInt(await input());
            if (i == 2) {
                print("     CHICKEN!!!!!\n");
                break;
            }
            n++;
            if (Math.random() > 0.833333) {
                print("     BANG!!!!!   YOU'RE DEAD!\n");
                print("CONDOLENCES WILL BE SENT TO YOUR RELATIVES.\n");
                break;
            }
            if (n > 10) {
                print("YOU WIN!!!!!\n");
                print("LET SOMEONE ELSE BLOW HIS BRAINS OUT.\n");
                restart = true;
                break;
            }
            print("- CLICK -\n");
            print("\n");
        }
        print("\n");
        print("\n");
        print("\n");
        print("...NEXT VICTIM...\n");
    }
}

main();

File: 76_Russian_Roulette/lua/russianroulette.lua
print [[
                    RUSSIAN ROULETTE
        CREATIVE COMPUTING  MORRISTOWN, NEW JERSY
This is a game of >>>>>>>>>>Russian Roulette
Here is a Revolver

]]

local function parse_input()
    local incorrect_input = true
    local input = nil
    while incorrect_input do
        input = io.read(1)
        if input == "1" or input == "2" then incorrect_input = false end
    end
    return input
end

local function russian_roulette()
    local NUMBER_OF_ROUNDS = 9

    while true do
        local dead = false
        local n = 0
        print("Type '1' to Spin chamber and pull trigger")
        print("Type '2' to Give up")
        print("Go")

        while not dead do
            local choice = parse_input()
            if choice == "2" then break end

            if math.random() > 0.833333333333334 then
                dead = true
            else
                print("CLICK")
                n = n + 1
            end

            if n > NUMBER_OF_ROUNDS then break end
        end

        if dead then
            print("BANG!!!!!   You're Dead!")
            print("Condolences will be sent to your relatives.\n\n\n")
            print("...Next victim...")
        elseif n > NUMBER_OF_ROUNDS then
            print("You win!!!!!")
            print("Let someone else blow his brain out.\n")
        else
            print("     Chicken!!!!!\n\n\n")
            print("...Next victim....")
        end
    end
end

russian_roulette()

